### PR TITLE
Major module update

### PR DESCRIPTION
**Pull request by jorgecotillo**
Updates contains:

1. Use of single diagnostic storage account
2. Enhanced Log Analytics -> added solutions and data sources
3. Added Automation Account
4. Added Update management based on tags
5. Resized VMs, min is DS2_V2
6. Enabled service endpoint (Storage account and Key Vault)

### CODE REVIEW

File: modules/kv/1.0/azureDeploy.json [hunk 1]
```diff
@@ -66,18 +66,16 @@
...
+        "la-resourceId": {
             "type": "string",
-            "minLength": 1,
             "metadata": {
-                "description": "Name of the Resource Group housing the OMS workspace."
+                "description": "Log Analytics workspace resource identifier"
             }
         },
-        "oms-workspace-name": {
+        "storageAccount-resourceId": {
```
Comment (senthilkungumaraj) [inline modules/kv/1.0/azureDeploy.json]: storageAccount-resourceId and la-resourceId pointing to the same resource id based on description. Can we combine it to one?

File: modules/kv/1.0/azureDeploy.json [hunk 2]
```diff
@@ -86,42 +84,38 @@
             "metadata": {
                 "description": "all secrets {\"secretName\":\"\",\"secretValue\":\"\"} wrapped in a secure object"
             }
+        },
+        "network-acls": {
+            "type": "object"
```
Comment (senthilkungumaraj) [inline modules/kv/1.0/azureDeploy.json]: missing description metadata

File: modules/storage-account-cdn/1.0/azureDeploy.json [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            "defaultValue": {
+                "name": "Standard_Microsoft"
+            }
+        },
+        "profileProperties": {
+            "type": "object",
+            "defaultValue": {}
+        },
+        "storageAccount-name": {
+            "type": "string"
```
Comment (senthilkungumaraj) [inline modules/storage-account-cdn/1.0/azureDeploy.json]: missing description metadata

File: modules/storage-account-cdn/1.0/azureDeploy.json [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        },
+        "sku": {
+            "type": "object",
+            "defaultValue": {
+                "name": "Standard_Microsoft"
+            }
+        },
+        "profileProperties": {
+            "type": "object",
+            "defaultValue": {}
```
Comment (senthilkungumaraj) [inline modules/storage-account-cdn/1.0/azureDeploy.json]: missing description metadata

File: modules/storage-account-cdn/1.0/azureDeploy.json [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        "deployment-prefix": {
+            "type": "string",
+            "metadata": {
+                "description": "Deployment prefix. E.g. Organization-DeploymentType"
+            }
+        },
+        "sku": {
+            "type": "object",
+            "defaultValue": {
+                "name": "Standard_Microsoft"
```
Comment (senthilkungumaraj) [inline modules/storage-account-cdn/1.0/azureDeploy.json]: missing description metadata

File: modules/storage-account/1.0/azureDeploy.json [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    "parameters": {
+        "deployment-prefix": {
+            "type": "string",
+            "metadata": {
+                "description": "Deployment prefix. E.g. Organization-DeploymentType"
+            }
+        },
+        "lock-for-deletion": {
+            "type": "bool",
+            "defaultValue": true
```
Comment (senthilkungumaraj) [inline modules/storage-account/1.0/azureDeploy.json]: missing description metadata

File: modules/storage-account/1.0/azureDeploy.json [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        "enable-network-acls": {
+            "type": "bool",
+            "defaultValue": false,
+            "metadata": {
+                "description": "Flag to enable network acls to whitelist IPs and subnets"
+            }
+        },
+        "network-acls": {
+            "type": "object",
+            "defaultValue": {}
```
Comment (senthilkungumaraj) [inline modules/storage-account/1.0/azureDeploy.json]: missing description metadata

File: modules/storage-account/2.0/azureDeploy.json [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    "parameters": {
+        "storage-account-name": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "Name of the storage account"
+            }
+        },
+        "storage-account-kind": {
+            "type": "string",
```
Comment (senthilkungumaraj) [inline modules/storage-account/2.0/azureDeploy.json]: missing description metadata

File: modules/storage-account/2.0/tests/storage-account.module.tests.ps1 [hunk 9]
```diff
@@ -0,0 +1,154 @@
+<#
```
Comment (senthilkungumaraj) [inline modules/storage-account/2.0/tests/storage-account.module.tests.ps1]: 2.0 module folders to be removed.

File: orchestration/models/resource_module.py [hunk 10]
```diff
@@ -46,6 +46,10 @@ def __init__(
 
             if 'source' in module:
                 self._source = ResourceSource(module['source'])
+            else:
+                # Let's create a
```
Comment (senthilkungumaraj) [inline orchestration/models/resource_module.py]: comments to be updated.

File: modules/kv/1.0/azureDeploy.json (hunk 1 above)
Comment (jorgecotillo) [reply modules/kv/1.0/azureDeploy.json]: Based on module dependencies, these two are different resources. I'll update the description of storageAccount-resourceId

File: modules/kv/1.0/azureDeploy.json (hunk 2 above)
Comment (jorgecotillo) [reply modules/kv/1.0/azureDeploy.json]: done

File: modules/storage-account-cdn/1.0/azureDeploy.json (hunk 3 above)
Comment (jorgecotillo) [reply modules/storage-account-cdn/1.0/azureDeploy.json]: done

File: modules/storage-account-cdn/1.0/azureDeploy.json (hunk 5 above)
Comment (jorgecotillo) [reply modules/storage-account-cdn/1.0/azureDeploy.json]: done

File: modules/storage-account/1.0/azureDeploy.json (hunk 7 above)
Comment (jorgecotillo) [reply modules/storage-account/1.0/azureDeploy.json]: done

File: modules/storage-account/2.0/azureDeploy.json (hunk 8 above)
Comment (jorgecotillo) [reply modules/storage-account/2.0/azureDeploy.json]: to be deleted

File: modules/storage-account/2.0/tests/storage-account.module.tests.ps1 (hunk 9 above)
Comment (jorgecotillo) [reply modules/storage-account/2.0/tests/storage-account.module.tests.ps1]: deleted

File: orchestration/models/resource_module.py (hunk 10 above)
Comment (jorgecotillo) [reply orchestration/models/resource_module.py]: Updated comments.

**Pull request merged**